backend/apps/authentication/serializers.py

```diff
@@ -59,6 +61,16 @@ def validate_email(self, value):
         if User.objects.filter(email=value).exists():
             raise serializers.ValidationError('This email is already registered')
         return value
+    
+    def validate_logo(self, value):
+        """ 
+        Validate logo file size and format
+        """
+        if value in (None, ''):
+            return value
+        validate_file_size(value, 10)
+        validate_file_type(value, ['image/jpeg', 'image/png'])
+        return value
```

Comment:
The serializer calls validate_file_size / validate_file_type which raise django.core.exceptions.ValidationError. DRF serializer field validation should raise rest_framework.serializers.ValidationError; raising Django's ValidationError here can produce uncaught exceptions or return 500 instead of proper 400 validation responses.

Catch Django ValidationError and re-raise as serializers.ValidationError, or change validator functions to raise serializers.ValidationError. 
```
from rest_framework import serializers as drf_serializers
...
try:
    validate_file_size(value, 10)
    validate_file_type(value, ['image/jpeg','image/png'])
except ValidationError as e:
    raise drf_serializers.ValidationError(str(e))
```